SparseGrids/tasgridUnitTests.cpp

```diff
@@ -786,4 +800,69 @@ std::vector<std::function<void(void)>> GridUnitTester::getRuntimeErrorCalls() co
     };
 }
 
+bool GridUnitTester::testLAPACKInterface() {
+
+    bool all_matched = true;
+    #ifdef Tasmanian_ENABLE_BLAS
+    // Initialize.
+    const int N = 100;
+    double* exact_eigs = new double[N];
+    double* D = new double[N];
+    double* E = new double[N-1];
+    double* W = new double[N];
+    double* Z = new double[N*N];
+    double* WORK1 = new double[2*N-2];
+    double* WORK2 = new double[4*N];
+    int* IBLOCK1 = new int[N];
+    int* ISPLIT1 = new int[N];
+    int* IWORK1 = new int[3*N];
+    double a = TasGrid::Maths::pi / 2;
+    double b = TasGrid::Maths::pi / 4;
+    ISPLIT1[0] = N;
+    IBLOCK1[0] = N;
+    for (int i=0; i<N; i++) {
+        exact_eigs[i] = a + 2.0 * b * std::cos((i+1) * TasGrid::Maths::pi / (N+1));
+    }
+    std::sort(exact_eigs, exact_eigs + N);
+
+    // Test LAPACK's dsterf function.
+    std::fill_n(D, N, a);
```

Comment:
Switching the above to `std::vector` will allow you to initialize the vector when you create it, then there is no need to call the `fill()` method.

Reply:
Done. I initialized D and E appropriately and removed the ``std::fill_n()``.